sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -154,6 +155,12 @@ void gnrc_lorawan_mcps_process_downlink(gnrc_lorawan_t *mac, uint8_t *psdu,
         return;
     }
 
+    /* Check if downlink was received after an uplink that had `ADRACKReq` bit set */
+    if (mac->mlme.adr_ack_cnt >= CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT) {
```

Comment:
The ADR ACK counter is reset always if a downlink is received, according to the standard (page 17)

Reply:
Done